extension/src/openvic-extension/singletons/CursorSingleton.cpp

```diff
@@ -0,0 +1,586 @@
+
+//, std::string_view const& base_folder
+String CursorSingleton::to_define_file_name(String const& path) const {
+	String name = path.replace("\\","/");
+	return name.get_slice("gfx/cursors/",1).get_slice(".",0);
+}
+
+bool CursorSingleton::load_cursors() {
+	GameSingleton const* game_singleton = GameSingleton::get_singleton();
+	ERR_FAIL_NULL_V_MSG(game_singleton, false, vformat("Error retrieving GameSingleton"));
+	
+	static constexpr std::string_view cursor_directory = "gfx/cursors";
+	bool ret = true;
+
+	//there is also a png file in the folder we don't want to bother loading
+	//so don't just load everything in the directory
+
+	//We need to load both ".cur" and ".CUR" files
+	Dataloader::path_vector_t cursor_files = game_singleton->get_dataloader()
+		.lookup_files_in_dir_recursive(cursor_directory, ".cur");
+
+	Dataloader::path_vector_t CURsor_files = game_singleton->get_dataloader()
+		.lookup_files_in_dir_recursive(cursor_directory, ".CUR");
+	cursor_files.insert(std::end(cursor_files),std::begin(CURsor_files),std::end(CURsor_files));
+
+	Dataloader::path_vector_t animated_cursor_files = game_singleton->get_dataloader()
+		.lookup_files_in_dir_recursive(cursor_directory, ".ani");
+	
+	if(cursor_files.size() < 1 && animated_cursor_files.size() < 1){
+		Logger::error("failed to load cursors: no files in cursors directory");
+		ret = false;
+	}
+
+	for(std::filesystem::path const& file_name : cursor_files) {
+		String file = std_to_godot_string(file_name.string());
+		String name = to_define_file_name(file);
+
+		if(!_load_cursor_cur(name,file)){
+			Logger::error("failed to load normal cursor at path ",file_name);
+			ret = false;
+		}
+		else{
+			cursor_names.append(name);
+		}
+	}
+	for(std::filesystem::path const& file_name : animated_cursor_files) {
+		String file = std_to_godot_string(file_name.string());
+		String name = to_define_file_name(file);
+
+		if(!_load_cursor_ani(name,file)){
+			Logger::error("failed to load animated cursor at path ",file_name);
+			ret = false;
+		}
+		else{
+			cursor_names.append(name);
+		}
+	}
+
+	return ret;
+}
+
+bool CursorSingleton::_load_cursor_ani(String const& name, String const& path) {
+	const Ref<FileAccess> file = FileAccess::open(path, FileAccess::ModeFlags::READ);
+
+	//todo make the size min(file's size, file's declared size) just like sound
+	//read the RIFF container
+	String riff_id = read_riff_str(file);
+	int riff_size = std::min(static_cast<uint64_t>(file->get_32()), file->get_length());;
+	String form_type = read_riff_str(file);
+
+	//important variables
+	std::vector<Array> frames_by_resolution;
+	std::vector<TypedArray<Vector2i>> hotspots_by_resolution;
+
+	TypedArray<Vector2i> resolutions;
+	TypedArray<float> displayRates;
+	TypedArray<int> sequence;
+
+	//ani header variables
+	int numFrames;
+	int numSteps;
+	Vector2i dimensions;
+	int bitCount;
+	int numPlanes; //???
+	int displayRate; //how long each frame should last
+	int flags;
+	bool iconFlag;
+	bool sequenceFlag;
```

Comment:
With MSVC primitive type variables aren't always default initialised, so these could contain random data. Always safer to explicitly initialise them, especially if there's any chance they'll be read later without having been changed.